src/test/java/edu/hm/hafner/analysis/parser/JsonParserTest.java

```diff
@@ -24,7 +24,8 @@ protected void assertThatIssuesArePresent(final Report report, final SoftAsserti
 
         softly.assertThat(report.get(0))
                 .hasFileName("test-file.txt")
-                .hasOnlyLineRanges(new LineRange(110, 111), new LineRange(120, 121))
```

Comment:
This check should still be working additionally to the other ones!

Reply:
You are right, I should better have read my own JavaDoc for the line ranges 🙈 

```
     * Sets additional line ranges for this issue. Not that the primary range given by {@code lineStart} and {@code *
     * lineEnd} is not included.
```

So in order to make this parser correct with respect to the API the test should have the assertions:
```
.hasOnlyLineRanges(new LineRange(120, 121))
.hasLineStart(110).hasLineEnd(111)
```

That means, your code in the parser should have an additional block right before building the issue (line 108) that removes the first line range if this range is stored in the line start and end attribute:

```
if (!lineRanges.isEmpty()) {
     LineRange firstRange = lineRanges.get(0);
     if (firstRange.getStart() == builder.getLineStart() && firstRange.getEnd() == builder.getLineEnd()) { 
           lineRanges.remove(0);
     }
}
```

Reply:
understood! I'll make the changes asap after my classes today

Reply:
also I have a question 

we're testing this
`.hasOnlyLineRanges(new LineRange(120, 121)) 
.hasLineStart(110)
.hasLineEnd(111)`

but, issues.json has both 110-111, and 120-121.

the test would inevitably fail with this condition

Reply:
Hmm, then it would be even better to move this code into the build method of the builder. There we have full access to all properties and there we can ensure that condition for all other parsers as well. 

So if issues contains the line ranges 110-111 and 120-121 then the build should return an issue with start and end lines set to 110 and 111. And one additional line range.